### DISCUSSION TITLE
Success with lg S09EQ.NSJ / S09EQ.UA3 !

### DISCUSSION THREAD

**Discussion by pieca2** [JanM321/esphome-lg-controller#42] [General]
I was looking for a way to automate my LG HVAC since a long time...
The LG website don't gives any compatibility list for their products with wifi boards or cable remote. The web doesn't seems to have many documentation on this either.
I was to the point to use an ESP32 fake infrared remote for this when I found this project.

An inspection to my HVAC mainboard reveals a CN-REMO port in it, so I ordered some PCB from JLPCB

While awaiting the PCB, I tried your code with an ESP32 and Home Assistant for a week with some automations and everything seems to work fluently. I'm even able to control the HVAC with Siri via HomeKit bridge.

Just got my tiny PCB from JLPCB today _(version February 24 without C3)_
I'm starring using this project  with your board connected to my HVAC and everything seems to work as promised, and without any annoying on/off or temperature change sounds from my unit !!!!

**Thank you so much to @florianbrede-ayet  and @JanM321  for all your work on this !!!!!**
One of the simplest and well documented project I've ever build.

This is my LG mono split model : **S09EQ.NSJ / S09EQ.UA3**
It is from 2018, but I had to open it to get access to the CN-REMO port on the PCB (no factory extension cable)

Just a thing :
For this project I've ordered a Giyofom ESP-WROOM-32E WiFi Dual Core CPU Development Board from Amazon.fr. This unit is narrower  than the tiny PCB (2 or 3mm). Bending the pins and forcing a little bit to fit solved the issue ;-)
![Untitled](https://github.com/JanM321/esphome-lg-controller/assets/165197185/efe5f4dc-7990-4393-93aa-a17117d9b9a7)

**Comment by JanM321** [comment]
Thanks for sharing this! It's always nice to hear these success stories 😄 

I've added the unit to the README.

**Reply by lagomCat** [reply]
Hi! I disassembled my S09EQR air conditioner and was glad to find a place on the board for the CN_REMO connector. However, I discovered that four transistors and several other components were not populated on the PCB.

That got me thinking — why go through the trouble of LIN bus conversion at all? The CN_REMO connector is used in some models for wired remote controls, and the transition from separate Tx/Rx lines to a single bidirectional line seems to be done to reduce the number of wires and possibly increase the cable length.

But if we’re already working directly with the internal board, it’s easy to trace the lines that go to the microcontroller’s Tx and Rx pins. In this case, wouldn't it be simpler to connect the ESP directly to these pins through a logic level shifter (5V ↔ 3.3V), bypassing the LIN transceiver entirely? In the photo: the blue dot is Tx, and the red dot is Rx.
![IMG_20250602_191404](https://github.com/user-attachments/assets/3a23ddb1-05b5-456c-b5db-32f032baead8)
